MekHQ/src/mekhq/campaign/mission/CrewSkillUpgrader.java

```diff
@@ -121,76 +126,76 @@ public void upgradeCrew(Entity entity) {
             xpCap -= spaCost;
         }
     }
-    
+
     /**
      * Upgrade an entity with a single SPA
      * @param entity
      * @return the xp cost of the added SPA
      */
-    public int addSingleSPA(Entity entity, double xpCap) {
+    private int addSingleSPA(Entity entity, double xpCap) {
         int unitType = entity.getUnitType();
-        
+
         List<SpecialAbility> choices = coalescedSPAList(unitType, xpCap);
```

Comment:
The main problem is likely that after the list of SPAs is built, the only SPAs left are ones that wind up ineligible. So I would have a loop in here that terminates either once we've picked an SPA or run out of SPAs to pick. Any time an SPA is eliminated below for whatever reason, remove it from the "choices" list and go through the loop again. That way, we're guaranteed to eventually reduces "choices" to an empty list or pick one that's valid.